File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = new HashSet<>();

        for (int num : nums) {
            if (set.contains(num)) {
                return true; // Duplicate found
            }
            set.add(num); // Add the number to the set
        }

        return false;
    }
}

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums.sort()
        start = 0
        end = 1

        while end < len(nums):
            if nums[start] == nums[end]:
                return True
            
            start +=1 
            end +=1

        return False